packages/adapter-auto/index.js

```diff
@@ -10,31 +13,39 @@ for (const candidate of adapters) {
 
 		try {
 			module = await import(candidate.module);
-
-			fn = () => {
-				const adapter = module.default();
-				return {
-					...adapter,
-					adapt: (builder) => {
-						builder.log.info(`Detected environment: ${candidate.name}. Using ${candidate.module}`);
-						return adapter.adapt(builder);
-					}
-				};
-			};
-
-			break;
 		} catch (error) {
 			if (
 				error.code === 'ERR_MODULE_NOT_FOUND' &&
 				error.message.startsWith(`Cannot find package '${candidate.module}'`)
 			) {
-				throw new Error(
-					`It looks like ${candidate.module} is not installed. Please install it and try building your project again.`
-				);
+				try {
+					execSync(
+						`echo "Installing ${candidate.module}" && npm install ${candidate.module} --no-save --no-package-lock`,
```

Comment:
TIL `--no-package-lock` — if that allows this to work across package managers, that'd be pretty sweet. 

Should we add a message along the lines of 'you should add this to your package.json for faster build times in future?'

Reply:
That'd only help people who look at the build logs, right? I wonder how many people that is.

Also, is there a reason for the `echo` to be part of the `exec` as well, instead of just a regular `console.log`?

If we get rid of that, can we do something with `execFileSync` like what we're doing [here](https://github.com/sveltejs/kit/blob/1798dad0b962b0aaca56c7aba9ae6a30b196d4d9/packages/migrate/migrations/routes/utils.js#L49) so we can completely avoid the shell's processing and escaping?

Reply:
What's the advantage to that? The current approach forwards the output to the main shell, so it _could_ help in case something goes wrong.

Reply:
> That'd only help people who look at the build logs, right? I wonder how many people that is.

a small number, but larger than zero

> What's the advantage to that? The current approach forwards the output to the main shell

We'd still pipe `stderr` and `stdout`. The advantage is that there's no caveats around `candidate.module` being some weird string that we need to deal with — npm package names _should_ be fine, but for an example of the category of bugs it prevents, see https://github.com/sveltejs/kit/pull/6129/files.

Also `execFileSync` should be slightly faster as it doesn't involve a shell

Reply:
tried switching to it, can't switch because windows can't execute npm without the `.cmd` ending in this case

Reply:
Can we have `process.platform === 'win32' ? 'npm.cmd' : 'npm'` as the first argument of `execFileSync`? Does that work?